instrumentation/apache-httpclient/apache-httpclient-2.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v2_0/AbstractCommonsHttpClientTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachehttpclient.v2_0;
+
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestOptions;
+import java.net.URI;
+import java.util.Map;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpConnectionManager;
+import org.apache.commons.httpclient.HttpMethod;
+import org.apache.commons.httpclient.MultiThreadedHttpConnectionManager;
+import org.apache.commons.httpclient.methods.DeleteMethod;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.methods.HeadMethod;
+import org.apache.commons.httpclient.methods.OptionsMethod;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.methods.PutMethod;
+import org.apache.commons.httpclient.methods.TraceMethod;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+abstract class AbstractCommonsHttpClientTest extends AbstractHttpClientTest<HttpMethod> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  private static final HttpConnectionManager connectionManager =
+      new MultiThreadedHttpConnectionManager();
+
+  private static final HttpClient client = new HttpClient(connectionManager);
+
+  private static final HttpClient clientWithReadTimeout = new HttpClient(connectionManager);
+
+  @BeforeAll
+  static void setUp() {
+    client.setConnectionTimeout((int) CONNECTION_TIMEOUT.toMillis());
+    clientWithReadTimeout.setTimeout((int) READ_TIMEOUT.toMillis());
+  }
```

Comment:
Original groovy code also sets connection timeout on the client with read timeout. To me this looks a bit weird. I would have done it the same way using a method that builds the http client as the groovy code had.